Sources/Sentry/SentryDebugMetaBuilder.m

```diff
@@ -0,0 +1,107 @@
+#import "SentryDebugMetaBuilder.h"
+#import "SentryCrashDynamicLinker.h"
+#import "SentryDebugMeta.h"
+#import "SentryHexAddressFormatter.h"
+#import "SentryLog.h"
+#import <Foundation/Foundation.h>
+
+@interface
+SentryDebugMetaBuilder ()
+
+@property (nonatomic, strong) id<SentryCrashBinaryImageProvider> binaryImageProvider;
+
+@end
+
+@implementation SentryDebugMetaBuilder
+
+- (id)initWithBinaryImageProvider:(id<SentryCrashBinaryImageProvider>)binaryImageProvider
+{
+    if (self = [super init]) {
+        self.binaryImageProvider = binaryImageProvider;
+    }
+    return self;
+}
+
+- (NSArray<SentryDebugMeta *> *)buildDebugMeta
+{
+    NSMutableArray<SentryDebugMeta *> *debugMetaArray = [NSMutableArray new];
+
+    NSInteger imageCount = [self.binaryImageProvider getImageCount];
+    for (NSInteger i = 0; i < imageCount; i++) {
+        SentryCrashBinaryImage image = [self.binaryImageProvider getBinaryImage:i];
+        SentryDebugMeta *debugMeta = [self fillDebugMetaFrom:image];
+        [debugMetaArray addObject:debugMeta];
+    }
+
+    return debugMetaArray;
+}
+
+- (SentryDebugMeta *)fillDebugMetaFrom:(SentryCrashBinaryImage)image
+{
+    SentryDebugMeta *debugMeta = [[SentryDebugMeta alloc] init];
+    debugMeta.uuid = [SentryDebugMetaBuilder convertUUID:image.uuid];
+    debugMeta.type = @"apple";
+
+    if (image.vmAddress > 0) {
+        NSNumber *imageVmAddress = [NSNumber numberWithUnsignedLongLong:image.vmAddress];
+        debugMeta.imageVmAddress = sentry_formatHexAddress(imageVmAddress);
+    }
+
+    NSNumber *imageAddress = [NSNumber numberWithUnsignedLongLong:image.address];
+    debugMeta.imageAddress = sentry_formatHexAddress(imageAddress);
+
+    debugMeta.imageSize = @(image.size);
+
+    if (nil != image.name) {
+        debugMeta.name = [[NSString alloc] initWithUTF8String:image.name];
+    }
+
+    return debugMeta;
+}
+
+/**
+ * Copied from SentryCrashReport.addUUIDElement, because we don't want to change SentryCrashReport.
+ */
++ (NSString *_Nullable)convertUUID:(const unsigned char *const)value
```

Comment:
H: I don't see `addUUIDElement` deleted from `SentryCrashReport`, can we extract to a class/Utils or something and reuse in both places? the code is pretty much similar but written in a different language and also not tested, a bug fixing on that class would require a fix here too.
`DRY`

Reply:
In the beginning, I didn't want to touch SentryCrashReport, because it's a kind of a black box for me. But you are right with `DRY`. I extracted a method.